keyserver/server.go

```diff
@@ -281,9 +307,23 @@ func resolveAndListen(addr *Address) (ln net.Listener,
 	}
 }
 
+func updatePerms(addr *Address) map[int]bool {
+	perms := make(map[int]bool)
+	perms[protocol.KeyLookupType] = true
+	perms[protocol.KeyLookupInEpochType] = true
+	perms[protocol.MonitoringType] = true
+	perms[protocol.RegistrationType] = addr.AllowRegistration
+	return perms
+}
+
 // Shutdown closes all of the server's connections and shuts down the server.
 func (server *ConiksServer) Shutdown() error {
 	close(server.stop)
+	if server.httpServer != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Can we move this scope to line 199 (https://github.com/coast-team/coniks-go/blob/92c82f41423983acc2e0d2e60fb0858527b5e6e2/keyserver/server.go#L199) as
```
go func() {
  <-server.stop
  // shutdown httpServer [...]
}()
```
and remove `httpServer` from `ConiksServer`?